### PR TITLE
initial prometheus-alertmanager

### PR DESCRIPTION
**Pull request by auhlig**
As discussed: Could you take a look @BugRoger ? Many thanks.

### CODE REVIEW

File: monitoring/charts/prometheus-alertmanager/templates/config.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        group_by: ['alertname']
+        group_wait: 10s
+        group_interval: 10s
+        repeat_interval: 1h
+        receiver: webhook_slack
+
+      receivers:
+      - name: webhook_slack
+        slack_configs:
+        - api_url: $WEBHOOK_URL
```
Comment (auhlig) [inline monitoring/charts/prometheus-alertmanager/templates/config.yaml]: Placeholder, as slack webhook_url contains a secret token. So I guess this should go internally

File: monitoring/charts/prometheus-collector/values.yaml [hunk 2]
```diff
@@ -2,3 +2,4 @@ image: prom/prometheus:v1.2.1
 retention: 1h0m0s
 memory_chunks: "262144"
 max_chunks_to_persist: "131072"
+alertmanager_url: "prometheus-alertmanager.{{ .Release.Namespace }}:9093"
```
Comment (auhlig) [inline monitoring/charts/prometheus-collector/values.yaml]: Will this work?

File: monitoring/charts/prometheus-alertmanager/templates/config.yaml (hunk 1 above)
Comment (BugRoger) [inline monitoring/charts/prometheus-alertmanager/templates/config.yaml]: "{{.Values.slack_webhook_url}}"?

File: monitoring/charts/prometheus-alertmanager/templates/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      containers:
+        - name: prometheus 
+          image: "{{.Values.image}}"
+          args:
+            - -config.file=/etc/prometheus/alertmanager.yml
+          ports:
+            - name: http 
+              containerPort: 9093
+          volumeMounts:
+            - name: data
```
Comment (BugRoger) [inline monitoring/charts/prometheus-alertmanager/templates/deployment.yaml]: copy/paste leftover or is there actually a data dir required for alertmanager?

File: monitoring/charts/prometheus-alertmanager/templates/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+              mountPath: /prometheus
+            - name: config
+              mountPath: /etc/prometheus
+      volumes:
+        - name: data
```
Comment (BugRoger) [inline monitoring/charts/prometheus-alertmanager/templates/deployment.yaml]: same as above. probably copied from prometheus-collector

File: monitoring/charts/prometheus-collector/templates/config.yaml [hunk 5]
```diff
@@ -9,12 +9,27 @@ data:
     aggregated:container_cpu_usage_seconds_total{} = sum(container_cpu_usage_seconds_total) by (id) keep_common
     aggregated:container_cpu_user_seconds_total{} = sum(container_cpu_user_seconds_total) by (id) keep_common
 
+  alert.rules: |
```
Comment (BugRoger) [inline monitoring/charts/prometheus-collector/templates/config.yaml]: shouldn't this be in the `prometheus-alertmanager` configmap?

File: monitoring/charts/prometheus-collector/values.yaml (hunk 2 above)
Comment (BugRoger) [inline monitoring/charts/prometheus-collector/values.yaml]: yes, looks good.

File: monitoring/charts/prometheus-alertmanager/templates/config.yaml (hunk 1 above)
Comment (BugRoger) [inline monitoring/charts/prometheus-alertmanager/templates/config.yaml]: `{{Values.slack_webhook_url}}`?

File: monitoring/charts/prometheus-collector/values.yaml (hunk 2 above)
Comment (BugRoger) [inline monitoring/charts/prometheus-collector/values.yaml]: yes, looks good

File: monitoring/charts/prometheus-collector/templates/config.yaml (hunk 5 above)
Comment (auhlig) [inline monitoring/charts/prometheus-collector/templates/config.yaml]: Alerting rules are configured and evaluated in the prometheus server. Subsequently alerts are sent to the alertmanager.

File: monitoring/charts/prometheus-alertmanager/templates/deployment.yaml (hunk 3 above)
Comment (auhlig) [inline monitoring/charts/prometheus-alertmanager/templates/deployment.yaml]: True. Data mountpoint isn't needed.

**Comment by auhlig** [comment]
TODO:

- [ ]  Certificate for Alertmanager
- [ ] verify slack integration

**Comment by BugRoger** [comment]
LGTM

Certificate is added as well.

**Comment by auhlig** [comment]
Thanks @BugRoger

**Pull request merged**